empress/support_files/js/util.js

```diff
@@ -100,6 +98,24 @@ define(["underscore"], function (_) {
         return result.concat(alphaPart, numericPart);
     }
 
+    /**
+     * Returns true if a string represents a finite number, false otherwise.
+     *
+     * Note that number parsing / type conversions in general are extremely
+     * hairy in JS. This function seems to work well for a wide range of cases,
+     * but it may not be perfect.
+     *
+     * This function was created from code that was originally in
+     * splitNumericValues() in Emperor's codebase. That code, in turn, was
+     * based on http://stackoverflow.com/a/9716488.
+     *
+     * @param {String} value Value to check for validity
+     * @return {Boolean} true if value represents a finite number, else false
+     */
+    function isValidNumber(value) {
+        return !isNaN(parseFloat(value)) && isFinite(value);
```

Comment:
```suggestion
        return !isNaN(parseFloat(value)) && isFinite(parseFloat(value));
```
This check will fail if value is "1/3".

Reply:
Weirdly enough, the recommended way to do this check seems to be without calling `parseFloat()` on the value passed to `isFinite()` -- see https://stackoverflow.com/a/9716488/10730311.

This function does return false if the input is `"1/3"`, but this is a good thing IMO: JavaScript does not parse fraction strings as expected --

```js
> parseFloat("1/3");
1
```

Changing the passed value to `isFinite()` to use `parseFloat()` causes `isValidNumber("1/3")` to be true, which is not desirable because we'd then interpret it as `1` misleadingly.

That all said, I thought it'd be good to add some unit tests in for `isValidNumber()` that directly check this functionality -- was working on that when I ran into those QUnit problems I mentioned on slack... once I get around those I'll update this.

Does this all sound reasonable?

Reply:
makes sense